distribution/lib/Standard/Base/0.0.0-dev/src/Data/Dictionary.enso

```diff
@@ -174,9 +174,9 @@ type Dictionary key value
     insert self key=(Missing_Argument.throw "key") value=(Missing_Argument.throw "value") no_warning:Boolean=False =
         new_dict = self.insert_builtin key value
         case key of
-            _ : Float ->
+            key_as_float : Float ->
                 if no_warning then new_dict else
-                    Warning.attach (Floating_Point_Equality.Used_As_Dictionary_Key key) new_dict
+                    Warning.attach (Floating_Point_Equality.Used_As_Dictionary_Key key_as_float) new_dict
```

Comment:
With intersection types, with
```
case a of
   b : T -> ...
```
The `a` and `b` are no longer interchangeable. The `T` part could be hidden in `a` (so you cannot pass it into functions that expect `T`) and it can be un-hidden by the `b : T` check at which point `b` has it 'visible' and `b` can be passed to `f (t : T)` whereas `a` cannot.

So we need to keep in mind that whenever we rely on `case of`, we should not do `_ : T` but name it and use the named component. At least anywhere where we may expect the intersection types to come up.

cc: @jdunkerley @GregoryTravis 

This is actually an important change to Enso semantics that we kind of knew about in #11600 but I'm not sure we have appreciate its implications enough yet.

Reply:
- known since https://github.com/enso-org/enso/pull/11850#discussion_r1883374897
- [documented](https://github.com/enso-org/enso/blob/32bdcebec4f4a4ac40eb8ba97af73855cf4bdd7b/docs/types/intersection-types.md#narrowing-type-check) as _"Remember to use `f.sqrt` and not `c.sqrt`."_

Reply:
Yes, I'm just reminding.